### ISSUE TITLE
Fix repodata for signed rpm/srpm/drpm fixtures

### ISSUE THREAD

**Issue by goosemania** [PulpQE/pulp-fixtures#32]
Right now sync fails if size/checksum verification is enabled ([validate option for importer is enabled ](http://docs.pulpproject.org/plugins/pulp_rpm/tech-reference/yum-plugins.html#configuration-parameters)).

Fixture repos that fail: 

- https://repos.fedorapeople.org/pulp/pulp/fixtures/drpm/

- https://repos.fedorapeople.org/pulp/pulp/fixtures/srpm/

For example, the following repos work:

-  https://repos.fedorapeople.org/repos/pulp/pulp/demo_repos/test_drpm_repo/

-  https://repos.fedorapeople.org/repos/pulp/pulp/demo_repos/test_srpm_repo/

**Comment by Ichimonji10** [comment]
My initial guess as to why this issue is occurring is that we generate repodata before signing files.

The correct solution for SRPMs is easy: just sign the SRPMs before generating the repodata. Hopefully, the solution for DRPMs will be the same. When solving this issue, bear in mind that the DRPM-generation tools always generate unsigned files, even if the input RPMs have identical signatures.

**Comment by Ichimonji10** [comment]
I think a good solution would be to update the fixture generation scripts to accept an optional argument, `--signing-key`. If passed, then the named key is used to sign the generated fixtures. This requires that a proper argument parsing library be used to parse arguments, instead of the current hack-job.

**Comment by Ichimonji10** [comment]
Upon further investigation, I've determined that this issue affects not only DRPM and SRPM repositories, but also RPM repositories.

The issue is that, when creating a signed repository, packages and metadata are written to the destination directory before package signing is performed. When packages are signed, the existing metadata is no longer accurate. Of interest to us is that package checksums and sizes are no longer accurate. Here's what a problematic make rule looks like:

```make
fixtures/rpm: gnupghome
	rpm/gen-fixtures.sh $@ rpm/assets
	GNUPGHOME=$$(realpath -e gnupghome) rpmsign --define '_gpg_name Pulp QE' \
		--addsign --fskpath ./rpm/GPG-RPM-PRIVATE-KEY-pulp-qe --signfiles \
		$$(find $@ -name '*.rpm')
```

I don't believe reverting to an earlier version of Pulp Fixtures would fix this issue. Formerly, all source packages were signed. Consequently, signed repositories had correct metadata. But when creating an unsigned repository, packages and metadata were written out before signatures were stripped from packages. Here's an example of what a make rule formerly looked like:

```make
fixtures/rpm-unsigned:
	rpm/gen-fixtures.sh $@ rpm/assets
	rpm/del-fixtures-sign.sh $@
```
I don't know why this didn't trigger any test failures. Perhaps the imports failed, Pulp didn't complain in an obvious way, and we weren't checking for sanity by re-downloading files?

I've renamed this issue to reflect what needs to be done.
